matchzoo/generators/point_generator.py

```diff
@@ -0,0 +1,62 @@
+"""Matchzoo point generator."""
+
+import numpy as np
+from matchzoo import engine
+from matchzoo import tasks
+from matchzoo.datapack import DataPack
+
+
+class PointGenerator(engine.BaseGenerator):
+    """PointGenerator for Matchzoo.
+
+    Ponit generator can be used for classification as well as ranking.
+
+    # Arguments
+        inputs: the transformed dataset
+    """
+
+    def __init__(
+        self,
+        inputs: DataPack,
+        task_type: engine.BaseTask,
+        batch_size: int=32,
+        shuffle: bool=True
+    ):
+        """Initialize the point generator."""
+        self.task_type = task_type
+        self.n = len(inputs)
+        self.transform_data(inputs)
+        super(PointGenerator, self).__init__(batch_size, shuffle)
+
+    def transform_data(self, inputs: DataPack):
+        """Obtain the transformed data from datapack."""
+        data = inputs.dataframe
+        self.x_left = np.asarray(data.text_left)
```

Comment:
Shouldn't add any attributes outside of `__init__`.

Reply:
So, can I assign the returned value to these attributes? just like `self.text_left, self.text_right = self.transform_data(inputs)`.